plugin/SpreadsheetPage.qml

```diff
@@ -18,81 +18,128 @@
 
 import QtQuick 2.0
 import Sailfish.Silica 1.0
+import Sailfish.Silica.private 1.0
 import org.kde.calligra 1.0 as Calligra
 
 DocumentPage {
     id: page
 
-    busy: doc.status != Calligra.DocumentStatus.Loaded
-    source: doc.source
-    indexCount: doc.indexCount
-
-    attachedPage: Component {
-        SpreadsheetListPage {
-            document: doc
-        }
-    }
-
     onStatusChanged: {
-        //Delay loading the document until the page has been activated.
         if (status == PageStatus.Active) {
-            doc.source = page.path
+            doc.source = page.source
         }
-
         //Reset the position when we change sheets
         if (status == PageStatus.Activating) {
-            documentFlickable.contentX = 0
-            documentFlickable.contentY = 0
+            flickable.contentX = 0
+            flickable.contentY = 0
+        }
+    }
+
+    busy: doc.status != Calligra.DocumentStatus.Loaded
+
+    FadeBlocker {
+        id: fadeBlocker
+        color: "white"
+        Binding {
+            when: fadeBlocker.fullscreen
+            target: __silica_applicationwindow_instance.__quickWindow
+            property: "color"
+            value: fadeBlocker.color
```

Comment:
Suppose it's somewhere along these lines that break spread sheet when app gets background and then foreground again. Black screen and eglErrors.

Reply:
Ok sounds like setting the color to clear the OpenGL context has issues. Will investigate.

Reply:
Fixed.